plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -139,28 +140,35 @@ export class EventsProcessor {
                 clearTimeout(timeout1)
             }
 
+            const team = await this.teamManager.fetchTeam(teamId)
+            if (!team) {
+                throw new Error(`No team found with ID ${teamId}. Can't ingest event.`)
+            }
+
             if (data['event'] === '$snapshot') {
```

Comment:
Plus a post-merge review that we're unnecessarily fetching the team for every event when only snapshot events care for it. We should be fetching the team inside this if